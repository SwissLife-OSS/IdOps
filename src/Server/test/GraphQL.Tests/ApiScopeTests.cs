using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using IdOps.Authorization;
using Snapshooter.Xunit;
using Squadron;
using Xunit;

namespace IdOps.GraphQL.Tests
{
    public class ApiScopeTests : TestHelper
        , IClassFixture<MongoResource>
    {
        public ApiScopeTests(MongoResource mongoResource) : base(mongoResource)
        {
        }

        [Fact]
        public async Task ValidRequest()
        {
            // arrange
            ICollection<string> tenants = new List<string>();
            tenants.Add("TestTenant");

            ITestRequestBuilder requestBuilder = TestRequestBuilder.New()
                .AddExecutor(await CreateSchema())
                .AddRequestFromFile("GetApiScopes")
                .AddVariableValue("tenants", tenants)
                .AddScope(AuthorizationPolicies.Names.ApiAccess);

            await InsertApiScopeIntoDB();

            // act
            IExecutionResult result = await requestBuilder.ExecuteAsync();

            // assert
            result.ToJson()
                .MatchSnapshot();
        }

        [Fact]
        public async Task GetApiScopes_OfWrongTenant_ResultIsEmpty()
        {
            // arrange
            ICollection<string> tenants = new List<string>();
            tenants.Add("Test");

            ITestRequestBuilder requestBuilder = TestRequestBuilder.New()
                .AddExecutor(await CreateSchema())
                .AddRequestFromFile("GetApiScopes")
                .AddVariableValue("tenants", tenants)
                .AddScope(AuthorizationPolicies.Names.ApiAccess);

            await InsertApiScopeIntoDB();

            // act
            IExecutionResult result = await requestBuilder.ExecuteAsync();

            // assert
            result.ToJson()
                .MatchSnapshot();
        }

        [Fact]
        public async Task GetApiScopes_ForUserWihtoutPermission_IsDenied()
        {
            // arrange
            ICollection<string> tenants = new List<string>();
            tenants.Add("TestTenant");

            ITestRequestBuilder requestBuilder = TestRequestBuilder.New()
                .AddExecutor(await CreateSchema())
                .AddRequestFromFile("GetApiScopes")
                .AddVariableValue("tenants", tenants);

            await InsertApiScopeIntoDB();

            // act
            IExecutionResult result = await requestBuilder.ExecuteAsync();

            // assert
            result.ToJson()
                .MatchSnapshot();
        }
    }
}
