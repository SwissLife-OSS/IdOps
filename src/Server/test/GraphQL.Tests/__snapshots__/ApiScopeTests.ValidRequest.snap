{
  "data": {
    "apiScopes": [
      {
        "id": "aaa375c78dfa4730a7a22f4a805c419a",
        "name": "scope",
        "tenant": "TestTenant",
        "enabled": true,
        "title": "scope"
      }
    ]
  }
}
